pipeline-adapters/mlrun-pipelines-kfp-common/src/mlrun_pipelines/common/ops.py

```diff
@@ -475,20 +483,39 @@ def format_summary_from_kfp_run(kfp_run, project=None):
                 mlrun.common.constants.MLRunInternalLabels.runner_pod,
             ],
         )
-        if step and step in dag:
-            dag[step]["run_uid"] = get_in(run, "metadata.uid")
-            dag[step]["kind"] = get_in(run, "metadata.labels.kind")
+        alternative_step_name = replace_last(step, f"-{run_name}", "")
```

Comment:
this looks very specific to the way how kfp/mlrun works (not sure who creates steps). Maybe worth adding comments with explanations for future self?